tests/collector/test_backend.py

```diff
@@ -92,3 +94,17 @@ def __init__(self):
 
     # And it should not have constructed the custom backend
     assert CountingCollector.CONSTRUCTED_COUNT == 2
+
+
+def test_context_managed_collector():
```

Comment:
Fwiw I could easily be convinced to drop this test. In reality it's not testing anything except language features (i.e. that decorators work and that `with` calls `__enter__` and `__exit__`. We could decorate the function with `pragma: no cover` instead.

Reply:
This test should stick around based on the discussion on `.coveragerc`, and has been updated to no longer use the decorator function.